Automattic/wp-calypso
# CI: New job for building the @automattic/wpcom-block-editor package

Part of #32044.

#### Changes proposed in this Pull Request

Add a new job to CircleCI in order to build the new `@automattic/wpcom-block-editor` package. The idea is to replicate the same build process for Masterbar Notifications, so we automatically build the package on each PR and store it as an artifact we can retrieve on WP.com to deploy a new version.

The job builds 2 versions of the package: minified and unminified. This is to ensure compliance of Jetpack with the [Guideline 8](https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#8-plugins-may-not-send-executable-code-via-third-party-systems) for the plugin directory. JP sites will load the unminified script if `SCRIPT_DEBUG` is enabled (see p7jreA-2gu-p2 #comment-1887).

#### Testing instructions

* Check the details of the `build-wpcom-block-editor` this PR:
<img width="768" alt="Screen Shot 2019-04-10 at 15 38 57" src="https://user-images.githubusercontent.com/1233880/55856983-e8cda200-5ba6-11e9-9d3c-235fc0b49fd4.png">
<img width="1073" alt="Screen Shot 2019-04-10 at 15 39 29" src="https://user-images.githubusercontent.com/1233880/55856989-ed925600-5ba6-11e9-8ba1-11419e9db48a.png">

* You can also trigger a new one by clicking on the "Rebuild" button:
<img width="250" alt="Screen Shot 2019-04-10 at 15 41 19" src="https://user-images.githubusercontent.com/1233880/55857093-2cc0a700-5ba7-11e9-9ac6-70638f2e9b89.png">

* Go to the Artifacts tab. You'll need to be logged in to CircleCI proper via GitHub for this tab to be visible.
* Make sure the both build files (minified and unminified) can be downloaded.

<img width="1349" alt="Screen Shot 2019-04-11 at 14 50 52" src="https://user-images.githubusercontent.com/1233880/55933709-45919100-5c69-11e9-9010-b02c5137216b.png">
